### PR TITLE
Don't just hang there

### PR DESCRIPTION
**Pull request by richafrank**
Propagate timeout as test error, instead of hanging on test clients after a "Timing out..." warning

### CODE REVIEW

**Review by jikquantopian — APPROVED** [review]
*shrug* lgtm. If it runs, :shipit:.

File: nose_gevented_multiprocess/nose_gevented_multiprocess.py [hunk 1]
```diff
@@ -534,14 +534,22 @@ def run_clients(number_of_clients, server_port, stop_event):
         if waited is stop_event:
             # Clean up by terminating the clients
             for client in clients:
-                client.wait(timeout=1)
-                log.info("Terminating client %s" % client)
+                log.info("Killing client %s" % client)
                 try:
-                    client.terminate()
+                    client.kill()
```
Comment (ssanderson) [inline nose_gevented_multiprocess/nose_gevented_multiprocess.py]: Do we want to try to allow the subprocesses to clean up gracefully before sending a SIGKILL? As is, this more or less guarantees that we're going to leak databases/temp files.

Comment (richafrank) [reply nose_gevented_multiprocess/nose_gevented_multiprocess.py]: Yep, had pushed an intermediate version of that commit. Meant to wait and terminate and then wait and kill.

**Comment by richafrank** [comment]
FYI In order to terminate both the spawned shell and its spawned worker, the [change grew a bit](https://github.com/quantopian/nose_gevent_multiprocess/pull/2/files#diff-776e7bea83ed5b7c3753b6400e33293aR571).

**Pull request merged**